src/azul/service/source_service.py

```diff
@@ -117,8 +123,32 @@ def _put(self, key: str, sources: list[AnyJSON]) -> None:
                 'N': str(self._now() + self.expiration)
             }
         }
-        self._dynamodb.put_item(TableName=self.table_name,
-                                Item=item)
+        self._log_dynamodb_request('PUT', Item=item)
+
+    def _log_dynamodb_request(self, method: str, **kwargs) -> dict:
+
+        def dynamo_request():
+            if method == 'GET':
+                return self._dynamodb.get_item
+            elif method == 'PUT':
+                return self._dynamodb.put_item
+            else:
+                assert False, method
+
+        log.debug('Making %s request to DynamoDb table %s with %s',
+                  method,
+                  self.table_name,
+                  trunc_ellipses(json_head(128 + 1, kwargs), 128))
+        dynamo_request = dynamo_request()
+        start = time()
+        response = dynamo_request(TableName=self.table_name, **kwargs)
+        log.debug('Got %s response after %.3fs from %s to DynamoDb table %s with response body %r',
+                  'OK' if type(response) is dict else 'no',
```

Comment:
"Got no response ... with response body ..." is paradoxical. If I read a log message that said "no response", I would think that it meant that the operation timed out and the client method never returned, so I don't think "no response" is the ideal word choice here. We _did_ get a response, just not the type we were expecting. Under what circumstances can these methods return a type other than `dict`? 

This method's return type also needs to updated, since it returns the `response` which (apparently) isn't always a `dict`.